File: src/main/java/javabot/operations/time/TimezonesIndonesia.java
package javabot.operations.time;

public class TimezonesIndonesia implements Timezones{

    private static Tri<String> timezones = new Tri<String>( );

    static {
        for( Timezones timezone: Timezones.values( ) ) {
            timezones.insert( timezone.name( ), timezone.getTimezone( ) );
        }
    }

    private enum Timezones {
        Jakarta( "Asia/Jakarta" ),
        Makassar( "Asia/Makassar" ),
        Jayapura( "Asia/Jayapura" );

        private String timezone;

        Timezones( String timezone ) {
            this.timezone = timezone;
        }

        public String getTimezone( ) {
            return timezone;
        }
    }

    @Override
    public String get( String province ) {
        return timezones.get( province );
    }

    @Override
    public String getCapital( ) {
        return Timezones.Jakarta.getTimezone( );
    }
}

File: src/main/kotlin/javabot/operations/time/TimezonesIndonesia.kt
package javabot.operations.time

public class TimezonesIndonesia : Timezones {

    private enum class Timezones(public val timezone: String) {
        Jakarta("Asia/Jakarta"),
        Makassar("Asia/Makassar"),
        Jayapura("Asia/Jayapura")
    }

    override fun get(province: String): String? {
        return timezones.get(province)
    }

    override fun getCapital(): String {
        return Timezones.Jakarta.timezone
    }

    companion object {

        private val timezones = Tri<String>()

        init {
            for (timezone in Timezones.values()) {
                timezones.insert(timezone.name, timezone.timezone)
            }
        }
    }
}
